Assets/BossRoom/Scripts/Client/UI/HeroActionBar.cs

```diff
@@ -147,18 +208,18 @@ void OnButtonClickedDown(int buttonID)
                 basicAction = ActionType.GeneralRevive;
             }
 
-            switch (buttonID)
+            switch (buttonType)
             {
-                case k_BasicActionButtonID: m_InputSender.RequestAction(basicAction, SkillTriggerStyle.UI); break;
-                case k_SpecialAction1ButtonID: m_InputSender.RequestAction(m_NetState.CharacterData.Skill2, SkillTriggerStyle.UI); break;
-                case k_SpecialAction2ButtonID: m_InputSender.RequestAction(m_NetState.CharacterData.Skill3, SkillTriggerStyle.UI); break;
-                default: throw new System.Exception($"Unknown button ID {buttonID}");
+                case ActionButtonType.BasicAction: m_InputSender.RequestAction(basicAction, SkillTriggerStyle.UI); break;
```

Comment:
This implementation does look good. But I was expecting that these switch cases would vanish.

Reply:
Oh, right. Here is that version. 

Since I added a non-readonly variable to the internal struct, I turned it into a class. This is just because it's less surprising to use. (We pass it as a function parameter, and this way the function can just change the variable without having to re-assign it to the dictionary, or pass with the ref keyword.) I think this is fine because they are allocated during `Awake()`, and are very tiny.